src/control/server/ctl_smd_rpc.go

```diff
@@ -445,11 +482,11 @@ func (svc *ControlService) SmdManage(ctx context.Context, req *ctlpb.SmdManageRe
 			if err != nil {
 				return nil, errors.Wrap(err, msg)
 			}
-			addManageRespIDOnFail(svc.log, devRes, devs[0])
+			addManageRespIDOnFail(svc.log, devRes, devs.getFirst())
 			devResults = append(devResults, devRes)
 		case *ctlpb.SmdManageReq_Led:
 			if len(devs) == 0 {
-				// TODO DAOS-12670: Enable all LEDs to be acted upon by default.
+				// Operate on all devices by default.
 				return nil, errors.New("led-manage request expects one or more IDs")
```

Comment:
This error message seems to not be adapted any more.
From what I understand, it should more indicate that no VMD devices was discovered.

Reply:
devs is part of the output from mapIDsToEngine(), as such the length should never be zero so this is a sanity check to prevent future code modification causing undesirable behaviour.